qutip/qip/circuit.py

```diff
@@ -60,6 +60,16 @@ def DisplaySVG(data, *args, **kwargs):
 
 __all__ = ['Gate', 'QubitCircuit']
 
+__single_qubit_gates = ["RX", "RY", "RZ", "SNOT", "SQRTNOT", "PHASEGATE",
+                        "X", "Y", "Z", "S", "T"]
+__para_gates = ["RX", "RY", "RZ", "CPHASE", "SWAPalpha", "PHASEGATE",
+                "GLOBALPHASE", "CRX", "CRY", "CRZ"]
+__ctrl_gates = ["CNOT", "CSIGN", "CRX", "CRY", "CRZ", "CY", "CZ",
+                "CS", "CT"]
+__swap_like = ["SWAP", "ISWAP", "SQRTISWAP", "SQRTSWAP", "BERKELEY",
+               "SWAPalpha"]
+__toffoli_like = ["TOFFOLI"]
+__fredkin_like = ["FREDKIN"]
```

Comment:
I grouped the gates for exception handling. Hopes that doesn't conflict with #1249 . I think it should be fine, since nothing is affected outside `Gate` class. If it works, I'll use the list to clean up `QubitCircuit`. It looks messy.

Reply:
It won't lead to any conflict, no worries. It is a big help grouping the gates!